js/src/views/Dapps/dapps.js

```diff
@@ -111,58 +88,49 @@ export default class Dapps extends Component {
     return (
       <div
         className={ styles.item }
-        key={ app.url }>
+        key={ app.id }>
         <Summary app={ app } />
       </div>
     );
   }
 
-  renderGlobalApps () {
-    const { globalApps } = this.state;
+  loadImages () {
+    const { available } = this.state;
+    const { dappReg } = Contracts.get();
 
-    return globalApps.map(this.renderApp);
+    return Promise.all(available.map((app) => dappReg.getImage(sha3(app.id))))
```

Comment:
Sorry wrong about each cycle, but we still need the sha3 to display somewhere - so either pre-calculate and add it to the structure or revert. The sha3 is the actual id that it is registered with on dappreg, cannot do without.